rewrite-maven/src/main/java/org/openrewrite/maven/search/ParentPomInsight.java

```diff
@@ -79,26 +91,42 @@ public Validated<Object> validate() {
-                    String artifactId = resolvedPom.getValue(tag.getChildValue("artifactId").orElse(null));
-                    if (matchesGlob(groupId, groupIdPattern) && matchesGlob(artifactId, artifactIdPattern)) {
-                        String parentVersion = resolvedPom.getValue(tag.getChildValue("version").orElse(null));
-                        if (version != null) {
-                            if (!Semver.validate(version, null).getValue()
-                                    .isValid(null, parentVersion)) {
-                                return t;
+                    MavenResolutionResult mrr = getResolutionResult();
+                    ResolvedPom resolvedPom = mrr.getPom();
+                    MavenPomDownloader mpd = new MavenPomDownloader(mrr.getProjectPoms(), ctx, mrr.getMavenSettings(), mrr.getActiveProfiles());
+
+                    Parent ancestor = mrr.getPom().getRequested().getParent();
+                    String relativePath = tag.getChildValue("relativePath").orElse(null);
+                    while (ancestor != null) {
+                        String groupId = ancestor.getGroupId();
+                        String artifactId = ancestor.getArtifactId();
+                        if (matchesGlob(groupId, groupIdPattern) && matchesGlob(artifactId, artifactIdPattern)) {
+                            String parentVersion = ancestor.getVersion();
+                            if (versionComparator == null || versionComparator.isValid(null, parentVersion)) {
+                                // Found a parent pom that matches the criteria
+                                inUse.insertRow(ctx, new ParentPomsInUse.Row(
+                                        resolvedPom.getArtifactId(), groupId, artifactId, parentVersion, relativePath));
+                                return SearchResult.found(t);
                             }
                         }
-                        // Found a parent pom that matches the criteria
-                        String relativePath = tag.getChildValue("relativePath").orElse(null);
-                        inUse.insertRow(ctx, new ParentPomsInUse.Row(
-                                resolvedPom.getArtifactId(), groupId, artifactId, parentVersion, relativePath));
-                        return SearchResult.found(t);
+                        if (Boolean.FALSE.equals(recursive)) {
+                            return t;
+                        }
+                        try {
+                            ResolvedPom ancestorPom = mpd.download(ancestor.getGav(), null, null, mrr.getPom().getRepositories())
+                                    .resolve(emptyList(), mpd, ctx);
+                            ancestor = ancestorPom.getRequested().getParent();
+                            relativePath = null;
+                        } catch (MavenDownloadingException e) {
+                            return e.warn(t);
+                        }
```

Comment:
I wonder if we should do anything here to break the while loop, just to ensure we don't get stuck when there's persisted failures to download a particular parent; for instance the following, but open to other suggestions.
```suggestion
                        } catch (MavenDownloadingException e) {
                            ancestor = null;
                            return e.warn(t);
                        }
```

Reply:
It’s a return statement already :sweat_smile: 

Note that I’m not familiar with this marker mechanism, I just noticed that’s how exceptions were handled in other places.

Reply:
You're absolutely right. I blame the fever I got from hiking through the rain. 😅